crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -1262,3 +1262,16 @@ where
         })
     }
 }
+
+/// Extracts the `PieceIndex` from a `RecordKey`.
+///
+/// SAFETY: The `RecordKey` must be a valid `PieceIndex` multihash.
+fn decode_piece_index_from_record_key(key: &RecordKey) -> PieceIndex {
+    let len = key.as_ref().len();
+    let s = len - PieceIndex::SIZE;
+
+    let mut piece_index_bytes = [0u8; PieceIndex::SIZE];
+    piece_index_bytes.copy_from_slice(&key.as_ref()[s..]);
+
+    PieceIndex::from_bytes(piece_index_bytes)
+}
```

Comment:
I really don't like this, but I understand why you did it and that we have tests, so it is fine to have for now. Please remove `SAFETY` comment though, it is not related to safety (no `unsafe` here).

Reply:
Actually, I don't like it either, but for now, I can't think of anything better.